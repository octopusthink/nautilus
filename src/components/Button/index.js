import { css } from '@emotion/core';
import invariant from 'invariant';
import PropTypes from 'prop-types';
import React from 'react';

export const Button = ({
  children,
  minimal,
  navigation,
  primary,
  ...otherProps
}) => {
  invariant(
    (minimal === true && primary === true) === false,
    'Cannot use `minimal` and `primary` props together on a <Button> component.',
  );

  let Component = 'button';
  if (navigation === true) {
    Component = 'a';
  }

  return (
    // See: https://github.com/yannickcr/eslint-plugin-react/issues/1555
    // eslint-disable-next-line react/button-has-type
    <Component
      css={css`
        font-family: -apple-system;
        font-size: 17px;
        padding: 12px 32px;
        outline: none;
        text-decoration: none;
        border: 2px solid #c62e7f;
        color: #c62e7f;
        background: #fff;
        border-radius: 8px;
        margin: 0 16px 16px;
        line-height: 1.412;
        transition: box-shadow 200ms;

        &:active {
          border-color: #973367;
          color: #973367;
        }

        &:focus {
          box-shadow: 0 0 1px 4px #F5A3CE;
          outline: none;
        }

        &::-moz-focus-inner {
          border: 0;
        }

        &:hover {
          box-shadow: 0 2px 0 0 #973367;
        }

        ${primary &&
          css`
            background: #c62e7f;
            color: #fff;

            &:active {
              background: #973367;
              color: #fff;
            }
          `}

        ${minimal &&
          css`
            border-top: 0;
            border-left: 0;
            border-right: 0;
            border-radius: 0;
            padding-left: 0;
            padding-right: 0;
          `}

        ${navigation &&
          css`
            &::after {
              content: ' â†’';
              display: inline;
            }
          `}

          &:disabled {
            border-color: #9f9f9f;
            color: #9f9f9f;

            &:hover {
                box-shadow: 0 2px 0 0 #6C6C6C;
            }

            &:active {
              background: #6C6C6C;
            }

            &:focus {
              box-shadow: 0 0 1px 4px #D4D4D4;
            }

            &${primary &&
              css`
                background: #9f9f9f;
                color: #fff;
              `}

      `}
      {...otherProps}
    >
      {children}
    </Component>
  );
};

Button.defaultProps = {
  children: undefined,
  disabled: false,
  minimal: false,
  navigation: false,
  primary: false,
  type: 'button',
};

Button.propTypes = {
  /** @ignore */
  children: PropTypes.node,
  /** Disables the button; this applies a disabled style but **does not disable any event handlers for the button**. Your `onClick` handlers should check for `props.disabled` to modify their behaviour accordingly. */
  disabled: PropTypes.bool,
  /** Decrease the visual prominence of this button in the UI. */
  minimal: PropTypes.bool,
  /** Sometimes, you need a button that's actually a link. We got you! This outputs a `react-router-dom` `<Link>` tag that looks (and largely behaves) like a `<Button>`, but can used as navigation. Setting this to `true` enables `Link` properties; see: https://github.com/ReactTraining/react-router/blob/master/packages/react-router-dom/docs/api/Link.md. */
  navigation: PropTypes.bool,
  /** Increase the visual prominence of this button in the UI. */
  primary: PropTypes.bool,
  /** HTML `type` attribute for the button. Defaults to `"button"`. */
  type: PropTypes.oneOf(['button', 'reset', 'submit']),
};

export default Button;
