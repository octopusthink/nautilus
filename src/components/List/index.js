import { css } from '@emotion/core';
import styled from '@emotion/styled';
import PropTypes from 'prop-types';
import React from 'react';

import { bodySmall, bodyMedium, bodyLarge } from 'themes/mixins';

export const List = ({
  children,
  large,
  small,
  inverse,
  dark,
  light,
  ...otherProps
}) => {
  return <ul {...otherProps}>{children}</ul>;
};

List.defaultProps = {
  children: undefined,
  large: false,
  small: false,
  inverse: false,
  dark: false,
  light: false,
};

List.propTypes = {
  /** @ignore */
  children: PropTypes.node,
  /** Increase the visual prominence of the paragraph. */
  large: PropTypes.bool,
  /** Decrease the visual prominence of the paragraph. */
  small: PropTypes.bool,
  /** Inverse text colour. Used for dark backgrounds. */
  inverse: PropTypes.bool,
  /** Darken text colour. */
  dark: PropTypes.bool,
  /** Lighten text colour. */
  light: PropTypes.bool,
};

export default styled(List)(({ dark, inverse, large, light, small, theme }) => {
  return css`
      color: ${theme.colors.text.default};
      margin: 0 0 ${theme.spacing.margin.m};
      ${small && bodySmall(theme)};
      ${!small && !large && bodyMedium(theme)};
      ${large && bodyLarge(theme)};
      ${light &&
        css`
          color: ${theme.colors.text.light};
        `}

      ${dark &&
        css`
          color: ${theme.colors.text.dark};
        `}

      ${inverse &&
        css`
          color: ${theme.colors.text.inverse};

          ${light &&
            css`
              color: ${theme.colors.text.inverseLight};
            `}
          ${dark &&
            css`
              color: ${theme.colors.text.inverseDark};
            `}
        `}
    `;
});
