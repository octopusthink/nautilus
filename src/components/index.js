export { default as Heading } from './Heading';
export { default as Paragraph } from './Paragraph';
