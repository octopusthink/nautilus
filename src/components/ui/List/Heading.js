import React from 'react';

export const ListHeading = (props) => {
  return <React.Fragment {...props} />;
};

export default ListHeading;
