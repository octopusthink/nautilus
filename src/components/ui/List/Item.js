import { css } from '@emotion/core';
import styled from '@emotion/styled';
import PropTypes from 'prop-types';
import React, { forwardRef } from 'react';

import { toUnits } from 'styles';

export const Item = forwardRef((props, ref) => {
  const { children, ...otherProps } = props;

  return (
    <li ref={ref} {...otherProps}>
      {children}
    </li>
  );
});

export const styles = (props) => {
  const { theme } = props;

  return css`
    position: relative;
    margin: 0 0 ${toUnits(theme.spacing.padding.extraSmall)};
    &::before {
      position: absolute;
      right: 101%;
    }
  `;
};

Item.defaultProps = {
  children: undefined,
};

Item.propTypes = {
  /** @ignore */
  children: PropTypes.node,
};

Item.displayName = 'List.Item';

export const { defaultProps, propTypes } = Item;

export default styled(Item)(styles);
