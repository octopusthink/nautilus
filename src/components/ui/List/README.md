A `<List>` is a collection of related elements of equal weight, presented in an itemised fashion.

Use the `<List.Item>` component for each item in your list.

```jsx
import { Heading } from '@octopusthink/nautilus';

<List ordered>
  <Heading>Things to do</Heading>
  <List.Item>Take shower. üõÄ</List.Item>
  <List.Item>Brush teeth. üòÅ</List.Item>
  <List.Item>Take over world. üåç</List.Item>
</List>
```

## Usage

Use a list to make blocks of text easier to read, and to break information into manageable chunks.

Don't use a `<List>` for:

- A collection of unrelated items. Use a `<Paragraph>` instead, or break the collection into smaller lists of related items.
- Presenting long or complex pieces of data. Use a `<Paragraph>` or a `<DataTable>` instead.
- A list of checkboxes, radio buttons, or other elements that require user interactivity. Use a `<Choice>` instead.
- A list of key/value pairs, such as a list of definitions or descriptions. Use a `<DescriptionList>` instead.

## Appearance

- Choose a bullet and numbering style and use them consistently across your site or app. Don't mix styles unless you have a good reason for doing so.
- Avoid lists of more than ten items. If you have a long list, consider breaking it up into smaller, more specific lists for more meaning and comprehension.
- Lists should use body text styling so they look identical to other body copy, especially when used in editorial copy. If a list stands alone, it can use more distinct typography.
- The space between each individual list item should be slightly bigger than the line-height inside the list item.
- Lists can be nested, but avoid too much nesting. As a general rule, nest lists to a maximum of three levels. If you need more levels, consider other ways of breaking up your content, like using an individual list for each top-level list item.
- Arrange list items in a logical way, even if they aren't sequential numbered steps. Bulleted lists can be arranged by priority, by order of importance, by size, alphabetically, numerically.

## Variations

Use a bulleted list when the order of items doesn't matter and to present content of equal status or value. Bulleted lists can be rearranged without losing any meaning.

```jsx
<List>
  <List.Item>cats</List.Item>
  <List.Item>dogs</List.Item>
  <List.Item>mice</List.Item>
</List>
```

<!--
TODO: Implement an Icon list.
Use an icon list when you want to provide additional semantic meaning‚Äîfor instance, to show dos and don'ts.

[example]
-->

Use ordered lists when item order is relevant, or to imply sequence and order. Ordered lists are commonly used when giving instructions or steps in a process.

```jsx
import { Paragraph } from '@octopusthink/nautilus';

<List ordered>
  <Paragraph>How to file taxes:</Paragraph>
  <List.Item>Obtain wine.</List.Item>
  <List.Item>Crack open wine.</List.Item>
  <List.Item>Start drinking.</List.Item>
  <List.Item>Spread papers in an organised fashion around room.</List.Item>
  <List.Item>Reluctantly do taxes.</List.Item>
</List>
```

## Voice & Tone

- Introduce lists with a lead-in `<Paragraph>` ending in a colon (eg: "My favourite things:") or a `<Heading>` with no punctuation.
- If list items use a single word only, start each item with a lowercase letter, and do not use a full stop at the end. (eg: "cat")
- If list items use full sentences or sentence fragments (subject and verb), start each item with an uppercase letter and use a full stop at the end. (eg: "The cat ate a mouse.")
- Never use commas or semicolons at the end of list items.
- Use list items that are grammatically parallel to reflect their equal weight. For example, do not mix passive voice with active voice or declarative statements with imperative commands, and follow the same phrase structure for each item in the list.

## Accessibility

Using lists well will help improve the accessibility of your site.

List items are conveyed as a group of related elements to assistive technology users. Assistive technologies may announce the total number of items in a list, give the relevant position of each item in a list, and provide shortcuts to quickly traverse a list.

- If your list has a title or lead-in phrase, use a `<Heading>` or `<Paragraph>` component, inside the list as the first component. This will identify it as the list header using the `aria-labelledby` attribute.
- Keep list labels as short as possible to communicate your intent.
- Use a bulleted list for lists where the order isn't important, and a ordered list when it is.
- Don't use a list for stylistic or presentation needs only.
- Use semantic HTML elements (`<ol>`, `<ul>`, and `<li>`) to mark up lists. (Nautilus will handle this for you!)

Here's how you should describe a list with a `<Heading>` component:

```jsx
import { Heading } from '@octopusthink/nautilus';

<List>
  <Heading level={4}>Tasty fruit</Heading>
  <List.Item>apples üçè</List.Item>
  <List.Item>kiwis ü•ù</List.Item>
  <List.Item>mangos ü•≠</List.Item>
</List>
```

Here's how you should describe a list with a `<Paragraph>` component:

```jsx
import { Paragraph } from '@octopusthink/nautilus';

<List ordered>
  <Paragraph>The best motorcycles, in order:</Paragraph>
  <List.Item>Honda CBR 125R</List.Item>
  <List.Item>Honda CB400 Four</List.Item>
  <List.Item>Ducati 848</List.Item>
</List>
```
