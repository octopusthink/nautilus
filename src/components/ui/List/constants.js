export const ListClassName = 'Nautilus-List';
export const ListItemClassName = 'Nautilus-ListItem';
