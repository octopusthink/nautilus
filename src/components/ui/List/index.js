import { css } from '@emotion/react';
import classnames from 'classnames';
import PropTypes from 'prop-types';
import React, { Children, cloneElement, useMemo, useState } from 'react';
import shortid from 'shortid';

import Heading from '../Heading';
import Paragraph from '../Paragraph';
import { bodyStyles } from '../../../styles';
import { useTheme } from '../../../themes';
import { ListClassName, ListItemClassName } from './constants';
import Item from './Item';

const List = (props) => {
  const {
    children,
    className,
    dark,
    inverse,
    large,
    light,
    noMargin,
    ordered,
    small,
    unstyled,
    ...otherProps
  } = props;

  let ListComponent = 'ul';
  if (ordered === true) {
    ListComponent = 'ol';
  }

  const [generatedId] = useState(shortid.generate());
  const description = useMemo(() => {
    const validDescriptionComponents = Children.toArray(children).filter((child) => {
      return child.type === Heading || child.type === Paragraph;
    });

    if (validDescriptionComponents.length === 0) {
      return undefined;
    }

    const descriptionComponent = validDescriptionComponents[0];

    if (!descriptionComponent.props.id) {
      return cloneElement(descriptionComponent, { id: generatedId });
    }

    return descriptionComponent;
  }, [children, generatedId]);

  const items = useMemo(() => {
    return Children.toArray(children).filter((child) => {
      return child.type !== Heading && child.type !== Paragraph;
    });
  }, [children]);

  const theme = useTheme();

  return (
    <React.Fragment>
      {description}
      <ListComponent
        aria-labelledby={description && description.props.id}
        css={
          unstyled
            ? undefined
            : css`
                ${bodyStyles({ dark, inverse, large, light, noMargin, small, theme })};
                padding: 0;
                ${ordered &&
                  css`
                    counter-reset: list-counter;

                    > .${ListItemClassName} {
                      list-style: none;
                      counter-increment: list-counter;

                      &::before {
                        content: counter(list-counter) '. ';
                      }
                    }
                  `}
                ${!ordered &&
                  css`
                    > .${ListItemClassName} {
                      list-style: none;

                      &::before {
                        content: '\\2022';
                        font-size: 0.6em;
                        line-height: 2.8;
                      }
                    }
                  `}
              `
        }
        className={classnames(ListClassName, className)}
        {...otherProps}
      >
        {items}
      </ListComponent>
    </React.Fragment>
  );
};

List.defaultProps = {
  children: undefined,
  className: undefined,
  dark: false,
  inverse: false,
  large: false,
  light: false,
  noMargin: false,
  ordered: false,
  small: false,
  unstyled: false,
};

List.propTypes = {
  /** @ignore */
  children: PropTypes.node,
  /** @ignore */
  className: PropTypes.string,
  /** Increase the visual prominence of the list. */
  large: PropTypes.bool,
  /** Decrease the visual prominence of the list. */
  small: PropTypes.bool,
  /** Inverse text colour. Used for dark backgrounds. */
  inverse: PropTypes.bool,
  /** Darken text colour. */
  dark: PropTypes.bool,
  /** Lighten text colour. */
  light: PropTypes.bool,
  /** Remove any outer margins from component. */
  noMargin: PropTypes.bool,
  /** Use numbers instead of bullets. */
  ordered: PropTypes.bool,
  /* @ignore Don't output any CSS styles. */
  unstyled: PropTypes.bool,
};

// Export Item as `List.Item`.
List.Item = Item;

export default List;
