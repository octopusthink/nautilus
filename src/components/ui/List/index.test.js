import React from 'react';

import { muteConsole, render } from 'utils/testing';

import { Heading, Paragraph } from 'components';
import List from '.';

describe('List', () => {
  it('should render a <ul> by default', () => {
    const { container } = render(<List />);

    expect(container.firstChild.tagName).toEqual('UL');
  });

  it('should render an <ol> when the ordered prop is true', () => {
    const { container } = render(<List ordered />);

    expect(container.firstChild.tagName).toEqual('OL');
  });

  it('should output its children', () => {
    const { getByTestId } = render(
      <List>
        <List.Item data-testid="child">hello</List.Item>
      </List>,
    );

    expect(getByTestId('child')).toBeDefined();
  });

  it('should not render non-List components as direct children', () => {
    muteConsole({ times: 1, type: 'error' });
    const { getByTestId } = render(
      <List>
        <div data-testid="child">hello</div>
      </List>,
    );

    expect(() => {
      getByTestId('child');
    }).toThrow('Unable to find an element');
  });

  it('should not render regular <li> tags as direct children', () => {
    muteConsole({ times: 1, type: 'error' });
    const { getByTestId } = render(
      <List>
        <li data-testid="child">hello</li>
      </List>,
    );

    expect(() => {
      getByTestId('child');
    }).toThrow('Unable to find an element');
  });

  it('should render several list items inside the list', () => {
    const { getByTestId } = render(
      <List>
        <List.Item data-testid="one">hello</List.Item>
        <List.Item data-testid="two">goodbye</List.Item>
      </List>,
    );

    expect(getByTestId('one').tagName).toEqual('LI');
    expect(getByTestId('two').tagName).toEqual('LI');
  });

  it('should ignore extra description components after the first ones', () => {
    const { getByTestId } = render(
      <List>
        <Heading level={3} data-testid="firstDescription">
          Tasty fruits
        </Heading>
        <Heading level={3} data-testid="secondDescription">
          Tasty fruits
        </Heading>
        <List.Item data-testid="one">apple</List.Item>
        <List.Item data-testid="two">banana</List.Item>
      </List>,
    );

    expect(getByTestId('firstDescription')).toBeDefined();
    expect(() => {
      getByTestId('secondDescription');
    }).toThrow('Unable to find an element');
  });

  it('should update the description component when it changes', () => {
    const { getByTestId, rerender } = render(
      <List>
        <Heading level={3} data-testid="firstDescription">
          Tasty fruits
        </Heading>
        <List.Item data-testid="one">apple</List.Item>
        <List.Item data-testid="two">raspberry</List.Item>
      </List>,
    );

    expect(getByTestId('firstDescription').tagName).toEqual('H3');

    rerender(
      <List>
        <Heading level={2} data-testid="firstDescription">
          Red fruits
        </Heading>
        <List.Item data-testid="one">apple</List.Item>
        <List.Item data-testid="two">raspberry</List.Item>
      </List>,
    );

    expect(getByTestId('firstDescription').tagName).toEqual('H2');
  });

  it('should not change the generated ID between renders', () => {
    const { getByTestId, rerender } = render(
      <List>
        <Heading level={3} data-testid="firstDescription">
          Tasty fruits
        </Heading>
        <List.Item data-testid="one">apple</List.Item>
        <List.Item data-testid="two">raspberry</List.Item>
      </List>,
    );

    const originalDescriptionId = getByTestId('firstDescription').getAttribute(
      'id',
    );

    rerender(
      <List>
        <Heading level={2} data-testid="firstDescription">
          Red fruits
        </Heading>
        <List.Item data-testid="one">apple</List.Item>
        <List.Item data-testid="two">raspberry</List.Item>
      </List>,
    );

    expect(originalDescriptionId).toEqual(
      getByTestId('firstDescription').getAttribute('id'),
    );
  });

  describe('described by Heading component', () => {
    it('should render a heading outside the list if one is provided', () => {
      const { container, getByTestId } = render(
        <List>
          <Heading level={3} id="fruits" data-testid="description">
            Tasty fruits
          </Heading>
        </List>,
      );

      expect(getByTestId('description').tagName).toEqual('H3');
      expect(container).toMatchSnapshot();
    });

    it('should set a random, short unique ID for the heading if no ID was provided', () => {
      const { getByTestId } = render(
        <List ordered>
          <Heading data-testid="description">My list</Heading>
          <List.Item>One</List.Item>
          <List.Item>Two</List.Item>
        </List>,
      );

      expect(getByTestId('description').getAttribute('id')).toBeDefined();
    });
  });

  describe('described by Paragraph component', () => {
    it('should render a paragraph outside the list if one is provided', () => {
      const { container, getByTestId } = render(
        <List>
          <Paragraph id="cars" data-testid="description">
            Here are some cars:
          </Paragraph>
          <List.Item>Honda Civic</List.Item>
          <List.Item>Ford Prefect</List.Item>
        </List>,
      );

      expect(getByTestId('description').tagName).toEqual('P');
      expect(container).toMatchSnapshot();
    });

    it('should set a random, short unique ID for the paragraph if no ID was provided', () => {
      const { getByTestId } = render(
        <List ordered>
          <Paragraph data-testid="description">My list</Paragraph>
          <List.Item>One</List.Item>
          <List.Item>Two</List.Item>
        </List>,
      );

      expect(getByTestId('description').getAttribute('id')).toBeDefined();
    });
  });
});
