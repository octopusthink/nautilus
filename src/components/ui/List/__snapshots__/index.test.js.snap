// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List described by Heading component should render a heading outside the list if one is provided 1`] = `
.emotion-1 {
  color: #3b3f45;
  margin: 0 0 3.2rem;
  font-family: -apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: 400;
  font-size: 1.8rem;
  line-height: 1.5555555555555556;
  padding: 0;
}

.emotion-1 > .Nautilus-ListItem {
  list-style: none;
}

.emotion-1 > .Nautilus-ListItem::before {
  content: '\\2022';
  font-size: 0.6em;
  line-height: 2.8;
}

.emotion-0 {
  margin: 0 0 3.2rem;
  font-family: -apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: 600;
  font-size: 3.1rem;
  line-height: 1.2903225806451613;
}

dl + .emotion-0,
.Nautilus-List + .emotion-0,
.Nautilus-Paragraph + .emotion-0 {
  margin-top: 3.2rem;
}

<div>
  <h3
    class="emotion-0"
    data-testid="description"
    id="fruits"
  >
    Tasty fruits
  </h3>
  <ul
    aria-labelledby="fruits"
    class="Nautilus-List emotion-1"
  />
</div>
`;

exports[`List described by Paragraph component should render a paragraph outside the list if one is provided 1`] = `
.emotion-3 {
  color: #3b3f45;
  margin: 0 0 3.2rem;
  font-family: -apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: 400;
  font-size: 1.8rem;
  line-height: 1.5555555555555556;
  padding: 0;
}

.emotion-3 > .Nautilus-ListItem {
  list-style: none;
}

.emotion-3 > .Nautilus-ListItem::before {
  content: '\\2022';
  font-size: 0.6em;
  line-height: 2.8;
}

.emotion-1 {
  position: relative;
  margin: 0 0 0.4rem;
}

.emotion-1::before {
  position: absolute;
  right: 101%;
}

.emotion-0 {
  color: #3b3f45;
  margin: 0 0 3.2rem;
  font-family: -apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: 400;
  font-size: 1.8rem;
  line-height: 1.5555555555555556;
}

<div>
  <p
    class="Nautilus-Paragraph emotion-0"
    data-testid="description"
    id="cars"
  >
    Here are some cars:
  </p>
  <ul
    aria-labelledby="cars"
    class="Nautilus-List emotion-3"
  >
    <li
      class="Nautilus-ListItem emotion-1"
    >
      Honda Civic
    </li>
    <li
      class="Nautilus-ListItem emotion-1"
    >
      Ford Prefect
    </li>
  </ul>
</div>
`;

exports[`List should not output styles when unstyled is set 1`] = `
.emotion-0 {
  position: relative;
  margin: 0 0 0.4rem;
}

.emotion-0::before {
  position: absolute;
  right: 101%;
}

<div>
  <ul
    class="Nautilus-List emotion-2"
  >
    <li
      class="Nautilus-ListItem emotion-0"
      data-testid="one"
    >
      apple
    </li>
    <li
      class="Nautilus-ListItem emotion-0"
      data-testid="two"
    >
      raspberry
    </li>
  </ul>
</div>
`;
